cnd/src/btsieve/bitcoin/mod.rs

```diff
@@ -8,142 +8,228 @@ pub use self::{
     transaction_pattern::TransactionPattern,
 };
 use crate::btsieve::{BlockByHash, LatestBlock, Predates};
+use crate::Never;
 use bitcoin::{
     consensus::{encode::deserialize, Decodable},
-    BitcoinHash,
+    BitcoinHash, OutPoint,
 };
 use chrono::NaiveDateTime;
 use futures_core::compat::Future01CompatExt;
+use genawaiter::sync::{Co, Gen};
+use genawaiter::GeneratorState;
 use reqwest::{Client, Url};
 use std::collections::HashSet;
 
-pub async fn matching_transaction<C>(
-    mut blockchain_connector: C,
-    pattern: TransactionPattern,
+type Hash = bitcoin::BlockHash;
+type Block = bitcoin::Block;
+
+// TODO: reevaluate naming of function
+pub async fn watch_for_transaction<C>(
+    blockchain_connector: C,
     start_of_swap: NaiveDateTime,
+    from_outpoint: OutPoint,
+    unlock_script: Vec<Vec<u8>>,
```

Comment:
Thinking ahead for miniscript, we might want to change this to be less generic and only provide a public key as the "spending with" condition.

i.e. we want to match whether it was spent with the redeem or the refund public key. To do that, I think we just have to check for the presence of the public key on the witness stack.

You can do this in a follow-up PR though if you want to keep the scope small.